endpoint/domain_filter.go

```diff
@@ -39,16 +46,26 @@ func prepareFilters(filters []string) []string {
 
 // NewDomainFilterWithExclusions returns a new DomainFilter, given a list of matches and exclusions
 func NewDomainFilterWithExclusions(domainFilters []string, excludeDomains []string) DomainFilter {
-	return DomainFilter{prepareFilters(domainFilters), prepareFilters(excludeDomains)}
+	return DomainFilter{prepareFilters(domainFilters), prepareFilters(excludeDomains), "", ""}
 }
 
 // NewDomainFilter returns a new DomainFilter given a comma separated list of domains
 func NewDomainFilter(domainFilters []string) DomainFilter {
-	return DomainFilter{prepareFilters(domainFilters), []string{}}
+	return DomainFilter{prepareFilters(domainFilters), []string{}, "", ""}
+}
+
+// NewRegexDomainFilter returns a new DomainFilter given a regular expression
+func NewRegexDomainFilter(regexDomainFilter string, regexDomainExclusion string) DomainFilter {
+	return DomainFilter{[]string{}, []string{}, regexDomainFilter, regexDomainExclusion}
```

Comment:
Can you please use named arguments for struct initialization instead of positional ones?

Reply:
I have changed that in the last commit, thanks :)